url8c14fd53ee_luca89_itaco_tgz-pJ8-main_RunnerJ8/src/main/Scanner.java

```diff
@@ -0,0 +1,989 @@
+			+ "\7\6\3\0\4\6\6\0\12\6\6\0\6\5\4\6\106\0\46\5"
+			+ "\12\0\51\5\7\0\132\5\5\0\104\5\5\0\122\5\6\0\7\5"
+			+ "\1\0\77\5\1\0\1\5\1\0\4\5\2\0\7\5\1\0\1\5"
+			+ "\1\0\4\5\2\0\47\5\1\0\1\5\1\0\4\5\2\0\37\5"
+			+ "\1\0\1\5\1\0\4\5\2\0\7\5\1\0\1\5\1\0\4\5"
+			+ "\2\0\7\5\1\0\7\5\1\0\27\5\1\0\37\5\1\0\1\5"
+			+ "\1\0\4\5\2\0\7\5\1\0\47\5\1\0\23\5\16\0\11\6"
+			+ "\56\0\125\5\14\0\u026c\5\2\0\10\5\12\0\32\5\5\0\113\5"
+			+ "\3\0\3\5\17\0\15\5\1\0\4\5\3\6\13\0\22\5\3\6"
+			+ "\13\0\22\5\2\6\14\0\15\5\1\0\3\5\1\0\2\6\14\0"
+			+ "\64\5\40\6\3\0\1\5\3\0\2\5\1\6\2\0\12\6\41\0"
+			+ "\3\6\2\0\12\6\6\0\130\5\10\0\51\5\1\6\126\0\35\5"
+			+ "\3\0\14\6\4\0\14\6\12\0\12\6\36\5\2\0\5\5\u038b\0"
+			+ "\154\5\224\0\234\5\4\0\132\5\6\0\26\5\2\0\6\5\2\0"
+			+ "\46\5\2\0\6\5\2\0\10\5\1\0\1\5\1\0\1\5\1\0"
+			+ "\1\5\1\0\37\5\2\0\65\5\1\0\7\5\1\0\1\5\3\0"
+			+ "\3\5\1\0\7\5\3\0\4\5\2\0\6\5\4\0\15\5\5\0"
+			+ "\3\5\1\0\7\5\17\0\4\6\32\0\5\6\20\0\2\5\23\0"
+			+ "\1\5\13\0\4\6\6\0\6\6\1\0\1\5\15\0\1\5\40\0"
+			+ "\22\5\36\0\15\6\4\0\1\6\3\0\6\6\27\0\1\5\4\0"
+			+ "\1\5\2\0\12\5\1\0\1\5\3\0\5\5\6\0\1\5\1\0"
+			+ "\1\5\1\0\1\5\1\0\4\5\1\0\3\5\1\0\7\5\3\0"
+			+ "\3\5\5\0\5\5\26\0\44\5\u0e81\0\3\5\31\0\11\5\6\6"
+			+ "\1\0\5\5\2\0\5\5\4\0\126\5\2\0\2\6\2\0\3\5"
+			+ "\1\0\137\5\5\0\50\5\4\0\136\5\21\0\30\5\70\0\20\5"
+			+ "\u0200\0\u19b6\5\112\0\u51a6\5\132\0\u048d\5\u0773\0\u2ba4\5\u215c\0\u012e\5"
+			+ "\2\0\73\5\225\0\7\5\14\0\5\5\5\0\1\5\1\6\12\5"
+			+ "\1\0\15\5\1\0\5\5\1\0\1\5\1\0\2\5\1\0\2\5"
+			+ "\1\0\154\5\41\0\u016b\5\22\0\100\5\2\0\66\5\50\0\15\5"
+			+ "\3\0\20\6\20\0\4\6\17\0\2\5\30\0\3\5\31\0\1\5"
+			+ "\6\0\5\5\1\0\207\5\2\0\1\6\4\0\1\5\13\0\12\6"
+			+ "\7\0\32\5\4\0\1\5\1\0\32\5\12\0\132\5\3\0\6\5"
+			+ "\2\0\6\5\2\0\6\5\2\0\3\5\3\0\2\5\3\0\2\5" + "\22\0\3\6\4\0";
+
+	/**
+	 * Translates characters to character classes
+	 */
+	private static final char[] ZZ_CMAP = zzUnpackCMap(ZZ_CMAP_PACKED);
+
+	/**
+	 * Translates DFA states to action switch labels.
+	 */
+	private static final int[] ZZ_ACTION = zzUnpackAction();
+
+	private static final String ZZ_ACTION_PACKED_0 = "\2\0\1\1\3\2\1\3\2\4\1\5\6\3\1\6"
+			+ "\1\7\1\10\1\11\1\12\1\13\1\14\1\15\1\16"
+			+ "\1\17\1\20\1\21\1\22\1\23\1\24\1\25\1\26"
+			+ "\2\27\1\30\1\1\3\3\1\31\4\3\1\32\1\33"
+			+ "\1\34\1\35\1\36\1\37\1\40\1\41\1\42\1\43"
+			+ "\17\3\1\44\6\3\1\45\1\46\1\47\4\3\1\50" + "\1\3\1\51\1\3\1\52";
+
+	private static int[] zzUnpackAction() {
+		int[] result = new int[89];
+		int offset = 0;
+		offset = zzUnpackAction(ZZ_ACTION_PACKED_0, offset, result);
+		return result;
+	}
+
+	private static int zzUnpackAction(String packed, int offset, int[] result) {
+		int i = 0; /* index in packed string */
+		int j = offset; /* index in unpacked array */
+		int l = packed.length();
+		while (i < l) {
+			int count = packed.charAt(i++);
+			int value = packed.charAt(i++);
+			do
+				result[j++] = value;
+			while (--count > 0);
+		}
+		return j;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 182:192, 296:307, 336:346.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.